ktlint-ruleset-standard/src/main/kotlin/com/pinterest/ktlint/ruleset/standard/rules/SpacingBetweenDeclarationsWithAnnotationsRule.kt

```diff
@@ -42,9 +42,8 @@ public class SpacingBetweenDeclarationsWithAnnotationsRule : StandardRule("spaci
         emit: (offset: Int, errorMessage: String, canBeAutoCorrected: Boolean) -> AutocorrectDecision,
     ) {
         node
-            .psi
-            ?.getPrevSiblingIgnoringWhitespaceAndComments(withItself = false)
-            ?.takeIf { it is KtDeclaration }
+            ?.getPrevSiblingIgnoringWhitespaceAndComments()
+            ?.takeIf { KtTokenSets.DECLARATION_TYPES.contains(it.elementType) }
```

Comment:
* Replace `contains` with `in`
* KtTokensSets is still declared in Psi (Java) library. It might be better to copy to our `TokensSets.kt`.

Reply:
It depends on the ultimate end goal. If the goal is to remove all references to `psi` than this would need to be refactored to. But that may be way to ambitious ;-)

We can leave it for now in Psi lib, as it is easy to refactor later.